components/prism-powerquery.js

```diff
@@ -0,0 +1,88 @@
+        /\bRecord\.(?:Combine|AddField|(?:Has|Remove|Rename|Reorder|Transform|Select)Fields|Field(?:Count|Names|OrDefault|Values)?|From(?:List|Table)|To(?:List|Table))\b/,
+        /\bReplacer\.(?:Replace(?:Text|Value))\b/,
+        /\bRowExpression\.(?:Column|From|Row)\b/,
+        /\bSplitter\.Split(?:ByNothing|TextBy(?:CharacterTransition|(?:Any|Each)?Delimiter|Lengths|Positions|Ranges|RepeatedLengths|Whitespace))\b/,
+        /\bTable\.(?:Add(?:(?:Index|Join)?Column|Key)|AggregateTableColumn|DuplicateColumn|AlternateRows|Buffer|Column(?:Count|Names|sOfType)?|Combine(?:Columns)?|Contains(?:All|Any)?|DemoteHeaders|Distinct|Expand(?:ListColumn|RecordColumn|TableColumn)|Fill(?:Down|Up)|FilterWithDataTable|FindText|First(?:N|Value)?|From(?:Columns|List|Partitions|Records|Rows|Value)|Fuzzy(?:Join|NestedJoin)|Group|HasColumns|InsertRows|IsDistinct|IsEmpty|Join|Keys|Last(?:N)?|Matches(?:AllRows|AnyRows)|Max|MaxN|Min|MinN|NestedJoin|Partition|Pivot|PositionOf(?:Any)?|PrefixColumns|Profile|PromoteHeaders|Range|Remove(?:Columns|FirstN|LastN|MatchingRows|Rows|RowsWithErrors)|RenameColumns|ReorderColumns|Repeat|Replace(?:ErrorValues|Keys|MatchingRows|RelationshipIdentity|Rows|Value)|ReverseRows|RowCount|Schema|Select(?:Columns|Rows|RowsWithErrors)|SingleRow|Skip|Sort|Split(?:Column)?|ToColumns|To(?:List|Records|Rows)|Transform(?:ColumnNames|Columns|ColumnTypes|Rows)|Transpose|Unpivot(?:OtherColumns)?|View(?:Function)?)\b/,
+        /\bTables\.GetRelationships\b/,
+        /\bText\.(?:At|(?:After|Before)Delimiter|BetweenDelimiters|Clean|Combine|Contains|End(?:sWith)?|From(?:Binary)?|InferNumberType|Insert|Length|Lower|Middle|NewGuid|Pad(?:End|Start)|PositionOf(?:Any)?|Proper|Range|Remove(?:Range)?|Repeat|Replace(?:Range)?|Reverse|Select|Split(?:Any)?|Start(?:sWith)?|To(?:Binary|List)|Trim(?:End|Start)?|Upper)\b/,
+        /\bTime\.(?:EndOfHour|From(?:Text)?|Hour|Minute|Second|StartOfHour|To(?:Record|Text))\b/,
+        /\bType\.(?:AddTableKey|ClosedRecord|Facets|For(?:Function|Record)|Function(?:Parameters|RequiredParameters|Return)|Is(?:Nullable|OpenRecord)?|ListItem|NonNullable|OpenRecord|RecordFields|Replace(?:Facets|TableKeys)|Table(?:Column|Keys|Row|Schema)|Union)\b/,
+        /\bUri\.(?:BuildQueryString|Combine|EscapeDataString|Parts)\b/,
+        /\bValue\.(?:Add|Compare|Divide|Equals|Metadata|Multiply|NativeQuery|NullableEquals|RemoveMetadata|ReplaceMetadata|Subtract|Type)\b/,
+        /\b(?:Byte|Currency|Decimal|Double|Guid|Int8|Int16|Int32|Int64|Percentage|Single)\.From\b/,
+        /\bJson.FromValue\b/
+    ],
+    'data-type':
+    {
+        pattern: /\b(?:any|anynonnull|binary|date|datetime|datetimezone|duration|function|list|logical|none|number|record|table|text|time|type)\b/,
+        alias: 'variable'
+    },
+    'number': /\b0x[\da-f]+\b|[+-]?\b\d+\.?\d*e?\d*|\B\.\d+\b/i,
```

Comment:
The [spec](https://docs.microsoft.com/en-us/powerquery-m/power-query-m-language-specification) has a nice grammar for number on pages 20-21.

```suggestion
    'number': /\b0x[\da-f]+\b|[+-]?(?:\b\d+\.|\B\.)?\d+(?:e[+-]?\d+)?/i,
```

Could you also add the following numbers to the numbers test, please?
```
123
123e45
123e+45
123e-45
123E45
123E+45
123E-45
123.78e-45
.78e-45
```

Reply:
That is much better, I made this change, but still there is one issue: This pattern above is now recognizing the double dot `..` operator. Example:
```
AA = .12,
BB = {1..100},
```
It recognizes 1.**.100** as a number (aka it includes one dot from the double dot operator in the number token.

I have tried to fix this, but I'm not so good with regex. As I see I cannot use `(?<!\.)` - could you maybe suggest a good solution pls?

Reply:
And also noticed that this pattern recognizes variable names as numbers if the have a digit in their names, like `MyVariable2`

Reply:
My bad. This pattern should work.

```js
{
    pattern: /\b0x[\da-f]+\b|(?:[+-]?(?:\b\d+\.)?\b\d+|[+-]\.\d+|(^|[^.])\B\.\d+)(?:e[+-]?\d+)?\b/,
    lookbehind: true
}
```

(Thank god, rewriting the `(?<!\.)` using a Prism lookbehind wasn't tedious at all... \s
Why did JS only add support for lookbehinds two years ago...)

Reply:
Super, thanks!